controllers/machinehealthcheck_controller.go

```diff
@@ -252,12 +254,57 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for _, t := range unhealthy {
 		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
 
-		if annotations.IsPaused(cluster, t.Machine) {
-			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+		if m.Spec.RemediationTemplate != nil {
+			// If external remediation request already exists,
+			// return early
+			if r.externalRemediationRequestExists(ctx, m, t) {
+				logger.Info("External Remediation Request for", t.Machine.Name, " exists")
+				return ctrl.Result{}, nil
+			}
+
+			cloneOwnerRef := &metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       t.Machine.Kind,
+				Name:       t.Machine.Name,
+				UID:        t.Machine.UID,
+			}
+
+			from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Remediation template not found %q", m.Spec.RemediationTemplate)
+			}
+
+			generateTemplateInput := &external.GenerateTemplateInput{
+				Template:    from,
+				TemplateRef: m.Spec.RemediationTemplate,
+				Namespace:   t.Machine.Namespace,
+				ClusterName: t.Machine.ClusterName,
+				OwnerRef:    cloneOwnerRef,
+				Labels:      nil,
+			}
+			to, err := external.GenerateTemplate(generateTemplateInput)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to create template for remediation request %q", m.Spec.RemediationTemplate)
+			}
+
+			to.SetName(t.Machine.Name)
```

Comment:
Let's add a comment that we require the remediation request to match the Machine name